### PR TITLE
Adding developer challenges and tidying up a few bits

### PR DESCRIPTION
**Pull request by CaisManai**

### CODE REVIEW

File: docs/community/journey.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+* Expect to lose all data when Testnet crashes or is restarted. 
+
+## Challenges
+
+Below are a set of challenges to get you started. For each challenge you successfully complete, you'll be rewarded task points which get transformed into OBX tokens as per the [Tokenomics](https://github.com/obscuronet/obscuro-project/wiki/Tokenomics)
+
+### Warmup - 20 points
+Get set up and deploy a simple contract.
+* Configure the Wallet extension and Metamask following the instructions [here](https://docs.obscu.ro/wallet-extension/wallet-extension.html)
+* Try deploying any contract (e.g. the default Storage contract). Remember to select Injected Provider as your Environment ![Step 1 screenshot](https://images.tango.us/public/screenshot_017a4a1c-b655-4eda-8c6e-7a298e1faa70.png?crop=focalpoint&fit=crop&fp-x=0.0945&fp-y=0.1827&fp-z=2.5115&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=2027%3A1103)
```
Comment (GavT) [inline docs/community/journey.md:42]: "If using Remix..." or something like "we recommend using Remix" or "this has been tested using Remix"

File: docs/community/journey.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+### Deploy the Guessing game - 20 points
+The guessing game is a very basic contract that showcases some of Obscuro's capabilities. The goal of the same is to simply guess the secret number the game has selected. Each time an attempt is made, an entrance fee of 1 token is paid. If a player guesses correctly, they win the entire pot and a new random number is selected.
```
Comment (GavT) [inline docs/community/journey.md:45]: typo: same >> game

File: docs/community/journey.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+## Challenges
+
+Below are a set of challenges to get you started. For each challenge you successfully complete, you'll be rewarded task points which get transformed into OBX tokens as per the [Tokenomics](https://github.com/obscuronet/obscuro-project/wiki/Tokenomics)
+
+### Warmup - 20 points
+Get set up and deploy a simple contract.
+* Configure the Wallet extension and Metamask following the instructions [here](https://docs.obscu.ro/wallet-extension/wallet-extension.html)
+* Try deploying any contract (e.g. the default Storage contract). Remember to select Injected Provider as your Environment ![Step 1 screenshot](https://images.tango.us/public/screenshot_017a4a1c-b655-4eda-8c6e-7a298e1faa70.png?crop=focalpoint&fit=crop&fp-x=0.0945&fp-y=0.1827&fp-z=2.5115&w=1200&mark-w=0.2&mark-pad=0&mark64=aHR0cHM6Ly9pbWFnZXMudGFuZ28udXMvc3RhdGljL21hZGUtd2l0aC10YW5nby13YXRlcm1hcmsucG5n&ar=2027%3A1103)
+
+### Deploy the Guessing game - 20 points
```
Comment (GavT) [inline docs/community/journey.md:44]: Should they get less points for deploying the guessing game then their own SC? They just need to follow the steps for the game but their own SC might blow up and need changes

File: docs/community/journey.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+The other is the *setNewTarget* function. This checks the game is in a state ready to be reset and then sets a secret number based on various block parameters.
+![setNewTarget function](../../assets/images/setnewtarget.png)
+
+### Extend the Guessing game with a proper win notification - 50 points
+You'll have noticed immediately that on winning the game, there is no notification! The only thing that happens is the game transfers to you all the tokens in the pot. 
+
+The challenge here is to extend the game with events (or anything else) to inform the winning player that they've won and how much they've won. 
+
+### Extend the Guessing game with Warmer/Colder functionality - 50 points
+Currently, after each guess, you have no idea how far you are from the winner number. The game will be more fun if players know with each play whether they're closer (warmer) or further (colder) form their last guess.
```
Comment (GavT) [inline docs/community/journey.md:64]: type: form >> from

File: docs/community/journey.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+There are two main functions of interest. First, the *attempt* function allows a player to guess the number and check if they've correctly guessed the secret number. If they have, the prize pot is paid out to them.
+![attempt function](../../assets/images/guessing.png)
+
+The other is the *setNewTarget* function. This checks the game is in a state ready to be reset and then sets a secret number based on various block parameters.
+![setNewTarget function](../../assets/images/setnewtarget.png)
+
+### Extend the Guessing game with a proper win notification - 50 points
+You'll have noticed immediately that on winning the game, there is no notification! The only thing that happens is the game transfers to you all the tokens in the pot. 
+
+The challenge here is to extend the game with events (or anything else) to inform the winning player that they've won and how much they've won.
```
Comment (GavT) [inline docs/community/journey.md:61]: Do you want to mention anything about implementing a good UI for this?

File: docs/wallet-extension/configure-metamask.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# Configuring MetaMask
```
Comment (GavT) [inline docs/wallet-extension/configure-metamask.md:1]: I thought we had a nice script which does this on the guessing game page? Can this be added to the configure-metamask page?

File: docs/wallet-extension/wallet-extension.md [hunk 7]
```diff
@@ -30,13 +30,7 @@ The diagram below lays out the flows.
...
 ## Usage
 
-1. You have two choices for getting the wallet extension:
-
-   a. Download the wallet extension binary for your operating system from the list of Assets in the [releases area on GitHub](https://github.com/obscuronet/go-obscuro/releases).
-   
-   Or
-
-   b. Compile the wallet extension binary by cloning the [Obscuro repository](https://github.com/obscuronet/go-obscuro)
+1. Compile the wallet extension binary by cloning the [Obscuro repository](https://github.com/obscuronet/go-obscuro)
```
Comment (GavT) [inline docs/wallet-extension/wallet-extension.md:33]: Why did you remove the download option? Its less friction.

**Review by GavT — COMMENTED** [review]
This is great! Some comments and questions left.

File: docs/community/journey.md (hunk 3 above)
Comment (CaisManai) [reply docs/community/journey.md:44]: The warm-up challenge isn't to write your own smart contract and deploy, but rather to use one of the default contracts that are shipped with Remix e.g. storage.sol

But we should add a final challenge: write your own Dapp

File: docs/community/journey.md (hunk 5 above)
Comment (CaisManai) [reply docs/community/journey.md:61]: Good point

File: docs/wallet-extension/configure-metamask.md (hunk 6 above)
Comment (CaisManai) [reply docs/wallet-extension/configure-metamask.md:1]: It was being repeated on multiple pages, so I pulled it out and made the steps simpler.

File: docs/wallet-extension/wallet-extension.md (hunk 7 above)
Comment (CaisManai) [reply docs/wallet-extension/wallet-extension.md:33]: James and Joel asked for it to be removed

**Review by GavT — APPROVED** [review]
LGTM
